### PR TITLE
[update] MIP-0 : Introduce status terms and editor role

### PR DESCRIPTION
**Pull request by apenzk**
# Summary

**Editors**. This PR introduces editors as an additional actor in the system and who takes responsibility to help with the process of MIPs.

**Status terms**. To identify the different stages already from the PR title, certain status terms are introduced.
 
There is also a need for an overview of existing MIPs. This PR is a prelude for the overview introduced in [this PR](https://github.com/movementlabsxyz/MIP/pull/31).

## Motivation

The distribution of MIP number seems confusing at times and errors may occur.  Contributors may feel hindered by the barrier of MIP-numbering. Furthermore there are a few steps such as correct naming of files or adding editors names to certain files that may not be obvious.

EIP has a well established standard : [https://eips.ethereum.org/](https://eips.ethereum.org/). However, the standard is too complex. Furthermore, the standard relies on a filtering process through a forum, which we do not have.

Inspired by EIP, this PR introduces a modification that can be handled in a small team, yet provides a clear path towards a well managed status of existing MIPs.

In addition it introduces the editor role to support the process.

### CODE REVIEW

**Review by franck44 — APPROVED** [review]

File: README.md
Comment (andygolay) [inline README.md:62]: @apenzk I like the statuses. Could you add more details about how the MIP number is assigned and registered? Perhaps there should be more about the editor role as well, if that's a new addition to our MIP process?

File: README.md
Comment (andygolay) [inline README.md:63]: One issue that may arise is: suppose an editor has a technical change that the author doesn't agree with. Is there some mechanism that we could add, to resolve such disagreements or disputes?

Comment (apenzk) [reply README.md:63]: (i have addressed now the second para in the text)

File: README.md
Comment (apenzk) [reply README.md]: I have removed the mention of the overview file, as it will be included by this [[update]](https://github.com/movementlabsxyz/MIP/pull/31/files)

i have added a note to clarify where the MIP number comes from. Essentially it will be fixed by the OVERVIEW file which comes with [this PR](https://github.com/movementlabsxyz/MIP/pull/31/files) from Franck.

File: README.md
Comment (andygolay) [reply README.md:67]: > > suppose an editor has a technical change that the author doesn't agree with. Is there some mechanism that we could add, to resolve such disagreements or disputes
> 
> How is this handled with code PRs?

For PRs, we have a minimum number of approvals needed (3 for MIPs, 2 for PRs to `aptos-core` and 1 for PRs to `movement`).

What about this solution: If an editor requests a change from an author, either the author can make the changes (tacitly approving the editorial feedback), or:

If the author doesn't agree with the changes, then the editor can mandate that the author implements the change suggestion by getting 2 upvotes on their discussion comment with the change requests (assuming change requests will be made in the Discussion section of the MIP PR)?

File: README.md
Comment (apenzk) [reply README.md]: > It may be helpful to define or introduce the editor role as well, if that's a new addition to our MIP process?

done.

File: README.md
Comment (apenzk) [reply README.md:67]: sounds good. added this. thx!

**Review by andygolay — APPROVED** [review]
This seems to be a good way to organize the MIPs and where they are in their lifecycles.

File: README.md
Comment (apenzk) [inline README.md:84]:
```suggestion
Currently the editors are [@apenzk](https://github.com/apenzk). Volunteers are welcome.
```

**Review by andygolay — CHANGES_REQUESTED** [review]

File: README.md
Comment (apenzk) [inline README.md:84]:
```suggestion
Currently the editors are [@apenzk](https://github.com/apenzk). 
```

Comment (apenzk) [reply README.md]: the intention was to invite company internal volunteers as opposed to external reviewers. i have removed the sentence as it was unclear.

Comment (apenzk) [reply README.md]: the editor can be bypassed as the rule remains that 3 approvals suffice. in which case the editor would have to apply [updates] if he is not satisfied. it is a non-blocking role. I have clarified this in the text.

Comment (apenzk) [reply README.md]: the point of [accepted] is that it gives the editor the chance to clean up the MIPs, ensure that MIP creators are added as code owners.

**Review by andygolay — COMMENTED** [review]
Changed review status to "comment" rather than "request changes"

File: README.md
Comment (andygolay) [reply README.md]: I'm not convinced anymore that the editor role and list are needed. If we just consider merged MIP PRs to be **Accepted** then the list of merged PRs can be our Accepted "list". Draft PRs are drafts and open PRs are in review or stale. 

The weekly MIP ceremonies can help us develop a discipline as a team to focus on relevant MIPs. Or in general if an MIP is falling by the wayside, we can ping other team members to review it. I don't think the editor role or a separate list are needed for this functionality.

Comment (apenzk) [reply README.md]: i have reduced the role of editor as well as removed the accepted stage. still think that the drafts are important and should be easy to oversee and which they currently are not.. 

here is the example i draw from at Ethereum https://eips.ethereum.org/core

Comment (andygolay) [reply README.md]: It makes sense in that case, because they're using a separate website apart from GitHub. I think if this level of extra administration is not strictly necessary, then it should be avoided.

Comment (apenzk) [reply README.md]: thx for the feedback though.. it really helps to understand what is actually needed.

File: README.md
Comment (l-monninger) [inline README.md:71]: I think the description on Withdrawn needs to be updated here.

File: README.md
Comment (l-monninger) [inline README.md:86]: I would add the note here that this is about shifting some work for the editor going forward. In particular, we want to add the authors to the CODEOWNERS such that they will be tagged on future revisions for a particular MD or MIP, e.g., errata.

File: README.md
Comment (l-monninger) [inline README.md:58]: Can you provide a justification for this as opposed to labels?

File: README.md
Comment (l-monninger) [inline README.md:64]: Good.

File: README.md
Comment (l-monninger) [inline README.md:66]: I might note in this very README that a big motivation for the role of the editor is the native readability of this format. We won't likely have something that re-renders the content herein for a while, so for people new to the format--particularly external users--the quality the editor ensures is important.

**Review by l-monninger — CHANGES_REQUESTED** [review]

File: README.md
Comment (l-monninger) [reply README.md]: @andygolay I would argue that because we will not be re-rerendering this content for a while, we should be taking extra care about its presentation natively.

Comment (andygolay) [reply README.md]: Agreed, with proper readability and quality standards in place that we're all aware of, it can be easy to implement.

**Review by andygolay — APPROVED** [review]

File: README.md
Comment (apenzk) [reply README.md:58]: There is no good reason for this as opposed to labels. I just compared with 
https://github.com/ethereum/EIPs/pulls and preferred it over not having anything. Would you prefer labels?

File: README.md
Comment (apenzk) [inline README.md:79]:
```suggestion
The motivation for the role of the editor is to ensure the readability and easy access of content, until further means, such as automatic rendering becomes available.

Currently the editors are [@apenzk](https://github.com/apenzk). 
```

File: README.md
Comment (apenzk) [reply README.md:66]: ok done at the bottom section

File: README.md
Comment (apenzk) [inline README.md:48]:
```suggestion
MIPs start as Drafts. MIPs obtain a number during the review process. The number is the highest number in the MIP and MD `overview.md` file plus 1. The editor can be requested for support on this.
```

File: README.md
Comment (apenzk) [reply README.md:64]: que?

File: README.md
Comment (apenzk) [inline README.md:71]:
```suggestion
- **Withdrawn** - an MIP that has not been revoked.
```

Comment (apenzk) [reply README.md:71]: done.

**Comment by apenzk** [comment]
@andygolay @l-monninger @franck44 
the numbering system will not be changed by this PR any longer. This author is not convinced any longer it improves the existing system. 
The main purpose - introduction of statuses and the editor role - remains.

File: README.md
Comment (apenzk) [reply README.md]: PR 33 has been merged and we reference here now to the relevant section

File: README.md
Comment (l-monninger) [reply README.md:54]: I think I would. If there is another opinion that supports mine, I would say we go with that.

File: README.md
Comment (l-monninger) [reply README.md:60]: This is a good thing to note.

**Review by l-monninger — APPROVED** [review]

**Comment by l-monninger** [comment]
Approved, but with the intent that GitHub labels might replace text-based labels if support is garnered.

**Pull request merged**
